sdk/python/kfp/registry/registry_client.py

```diff
@@ -152,73 +157,139 @@ def _validate_tag(self, tag: str) -> None:
-            package_endpoint = f'{api_endpoint}/packages'
-            package_name_endpoint = f'{package_endpoint}/{{package_name}}'
-            tags_endpoint = f'{package_name_endpoint}/tags'
-            versions_endpoint = f'{package_name_endpoint}/versions'
-            config = {
-                'host':
-                    self._host,
-                'upload_url':
-                    self._host,
-                'download_version_url':
-                    f'{self._host}/{{package_name}}/{{version}}',
-                'download_tag_url':
-                    f'{self._host}/{{package_name}}/{{tag}}',
-                'get_package_url':
-                    f'{package_name_endpoint}',
-                'list_packages_url':
-                    package_endpoint,
-                'delete_package_url':
-                    f'{package_name_endpoint}',
-                'get_tag_url':
-                    f'{tags_endpoint}/{{tag}}',
-                'list_tags_url':
-                    f'{tags_endpoint}',
-                'delete_tag_url':
-                    f'{tags_endpoint}/{{tag}}',
-                'create_tag_url':
-                    f'{tags_endpoint}?tagId={{tag}}',
-                'update_tag_url':
-                    f'{tags_endpoint}/{{tag}}?updateMask=version',
-                'get_version_url':
-                    f'{versions_endpoint}/{{version}}',
-                'list_versions_url':
-                    f'{versions_endpoint}',
-                'delete_version_url':
-                    f'{versions_endpoint}/{{version}}',
-                'package_format':
-                    f'{repo_resource_format}/packages/{{package_name}}',
-                'tag_format':
-                    f'{repo_resource_format}/packages/{{package_name}}/tags/{{tag}}',
-                'version_format':
-                    f'{repo_resource_format}/packages/{{package_name}}/versions/{{version}}',
-            }
         else:
-            logging.info(f'load_config not implemented for host: {self._host}')
+            if 'regex' in config:
+                matched = re.match(config['regex'], self._host)
+
+        if not bool(matched):
+            raise ValueError(f'Invalid host URL: {self._host}.')
+
+        # Replace all currently known variables with values
+        map_dict = _SafeDict(**matched.groupdict(), host=self._host)
+        for config_key in config:
+            config[config_key] = config[config_key].format_map(map_dict)
+
         return config
 
+    def _load_context(self,
+                      config_file: str,
+                      config: Optional[dict] = None) -> dict:
+        """Loads the context from the given config_file.
+
+        Args:
+            config_file: The location of the config file.
+            config: Optional. An existing config to set as the default config.
+
+        Returns:
+            The loaded config.
+        """
+        loaded_config = {}
```

Comment:
Can this line be removed? `loaded_config` is always reassigned in line 286 before it is referenced.

Reply:
Done.